misc/error_injector/error_injector.h

```diff
@@ -0,0 +1,84 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+ */
+
+#ifndef CELIX_ERROR_INJECTOR_H
+#define CELIX_ERROR_INJECTOR_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <assert.h>
+#include <dlfcn.h>
+#include <stddef.h>
+
```

Comment:
Could this macro get some documentation. Something like:

```
/**
 * @brief Get the caller library address based on the return address of level frame up t he call stack.
 */
#define CELIX_EI_GET_CALLER(addr, level)                                   \
do {                                                                       \
    Dl_info dlinfo;                                                        \
    dladdr(__builtin_return_address(level), &dlinfo);                     \
    (addr) = dlinfo.dli_saddr;                                             \
} while(0)
```

Reply:
Or maybe a small readme how to use this (declare, define, etc)

Reply:
I think both are needed.